Cargo.toml

```diff
@@ -32,9 +32,13 @@ trybuild = { version = "1.0.99", features = ["diff"] }
 default = ["syn-error"]
 syn-error = ["dep:syn"]
 nightly = []
+detect-nightly = []
```

Comment:
Is there a good reason not to simplify things by getting rid of both these feature flags and always using the build script?

There is *a* reason: to ensure compatibility with future nightly `rustc` using an incompatible reporting interface. This seems a moot point since (a) a fix can likely be published quickly as a patch release (no one supports old nightlies thus there will not be any supported working configuration to break), (b) `stable` will still work and (c) it shouldn't be that hard to work around using [`[patch.crates-io.proc-macro-error-2]`](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section).

Otherwise...

1. The existing `nightly` feature needs to be forwarded from whatever proc-macro lib uses this crate, and that lib needs to tell its users to add their own feature forwarding `nightly` and to use this when developing, as well as to add a CI test which uses this feature and denies warnings. This path is theoretically workable but has a lot of overhead.
2. This `detect-nightly` feature needs to be forwarded from whatever proc-macro lib uses this crate, and that lib needs to tell its users to enable it (either permanently or with their own feature flag used in testing as above). Since even the crate using the proc-macro crate which uses this crate is probably itself a library, it still is likely in the same situation as this crate about what it may assume.
3. The proc-macro built from this crate may permanently enable `detect-nightly` since this crate doesn't.

Further, the crate docs still seem to imply that warnings happen on nightly automatically.

My take: both these feature flags should be removed and detection should be automatic.

Reply:
> Further, the crate docs still seem to imply that warnings happen on nightly automatically.

That's because this crate is forked from `proc-macro-error` which had build scripts for nightly detection. The authors of this crate removed the scripts on the path to `syn2`.

I think the authors of this repo should list out their goals with the project. Namely where does compatibility with pme1 stack up to our own thoughts on build systems? `proc-macro-error` used a build script and we're only here because folks are still using it.

Reply:
> My take: both these feature flags should be removed and detection should be automatic.

I agree.

Reply:
I'm here because (1) this was an easy transition and (2) the alternatives did not support reporting-as-a-side-effect (`emit_*!`).

So, yes. And I'm not so happy to see that warnings are no longer being reported (without extra set up on my end that I didn't know I had to do).

The best option would be rustc stabilising https://github.com/rust-lang/rust/issues/54140 or equivalent. The next best option is something like `proc-macro-error`.